async-query-core/src/main/java/org/opensearch/sql/spark/utils/SQLQueryUtils.java

```diff
@@ -73,6 +75,32 @@ public static boolean isFlintExtensionQuery(String sqlQuery) {
     }
   }
 
+  public static List<String> validateSparkSqlQuery(String sqlQuery) {
+    SparkSqlValidatorVisitor sparkSqlValidatorVisitor = new SparkSqlValidatorVisitor();
+    SqlBaseParser sqlBaseParser =
+        new SqlBaseParser(
+            new CommonTokenStream(new SqlBaseLexer(new CaseInsensitiveCharStream(sqlQuery))));
```

Comment:
Not sure the overhead. We may want to do this once in future because Spark grammar is huge.

Reply:
https://github.com/opensearch-project/sql/issues/2886